astropy/coordinates/representation.py

```diff
@@ -221,6 +227,136 @@ def _unitstr(self):
                            for component in self.components]))
         return unitstr
 
+    def __eq__(self, other):
+        """Check for equality with another representation.
+
+        If either ``self`` or ``other`` are not scalar, this returns a boolean
+        array, just like is the case for ``ndarray``.  Elements are `True` only
+        if all components of that element are exactly equal.
+        """
+        try:
+            # this ensures that other can be compared sensibly to self.
+            # note that it immediately returns if the classes are the same.
+            other = other.represent_as(self.__class__)
+        except Exception:
+            return False
+
+        # Ensure we return a bool or array of bool as appropriate.
+        return functools.reduce(np.logical_and,
+                                (getattr(self, component) ==
+                                 getattr(other, component)
+                                 for component in self.components))
+
+    def __ne__(self, other):
+        return np.logical_not(self.__eq__(other))
+
+    def __mul__(self, other):
+        values = (getattr(self, component) for component in self.components)
+        return self.__class__(*(
+            (value if issubclass(cls, Angle) else value * other)
+            for (component, cls), value in zip(self.attr_classes.items(),
```

Comment:
You're right that these could be unified. I now did and decided to use a plain loop rather than list comprehension, which should make it a bit easier to follow (and allows me to pass back a `NotImplemented` in the process, in case `other` is a filehandle or so...).